src/webapp/js/components/modules-form.js

```diff
@@ -353,19 +355,22 @@ function validateFields(fields, values) {
   const errors = {};
 
   _.each(fields, (fieldConfig, fieldName) => {
+    const fieldValue = values[fieldName];
+
     if (fieldConfig.type === 'section') {
-      errors[fieldName] = validateFields(fieldConfig.fields, values[fieldName]);
+      errors[fieldName] = validateFields(fieldConfig.fields, fieldValue);
     } else if (fieldConfig.type === 'array') {
-      const array = values[fieldName];
-
-      if (array) {
+      if (fieldValue) {
         errors[fieldName] = [];
-        _.each(array, (value) => {
+        _.each(fieldValue, (value) => {
           errors[fieldName].push(validateFields(fieldConfig.fields, value));
         });
       }
-    } else if (fieldConfig.required && (values[fieldName] === undefined || values[fieldName] === null || values[fieldName] === '')) {
+    } else if (fieldConfig.required && (fieldValue === undefined || fieldValue === null || fieldValue === '')) {
```

Comment:
Instead of this 'triple or', you can use !!fieldValue here. It will return false if it will be undefined, null or empty., so in your case: ! ( !!fieldValue )

Reply:
it won't work if field value will be 0

Reply:
true, then it is ok.